webapp/appModules/data/records/components/recordView.js

```diff
@@ -45,19 +44,22 @@ class RecordView extends React.Component {
   }
 
   render () {
-    const {recordLoaded, preview} = this.props
+    const {recordLoaded, preview, user, record} = this.props
 
     return recordLoaded
-      ? <SurveyFormView draft={preview} preview={preview} edit={false} entry={true}/>
+      ? <SurveyFormView draft={preview} preview={preview} edit={false} entry={true} canEditRecord={canEditRecord(user, record)}/>
```

Comment:
do not pass user and record to props, calculate "canEditRecord" prop in mapStateToProps

Reply:
user and   record are still in mapStateToProps, do not pass them to props,